src/magic.cpp

```diff
@@ -603,15 +603,15 @@ bool spell::is_max_level() const
     return get_level() >= type->max_level;
 }
 
-bool spell::can_learn( const player &p ) const
+bool spell::can_learn( const Character &p ) const
```

Comment:
if you're going to change these lines, can you please change the parameters to multiple character variables? i've been using `guy` for character in my PRs. it is far easier to search for a variable that is more than one character. pun intended.